packages/cockroach/build

```diff
@@ -1,36 +1,57 @@
  #!/bin/bash
 
-set -e
+set -ex
 
-# cockroachdb requires gcc 4.9+
+# Support https sources in apt
 apt-get -y update
-apt-get install -y gcc-4.9 g++-4.9 cpp-4.9
+apt-get -y install apt-transport-https software-properties-common
 
-update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.9 40
-update-alternatives --install /usr/bin/cpp cpp /usr/bin/cpp-4.9 40
-update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.9 40
+# CockroachDB requires node and yarn to build UI
+# https://github.com/cockroachdb/cockroach/blob/v2.0.7/build/bootstrap/bootstrap-debian.sh#L8-L12
+curl -fsSL https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
+echo "deb https://deb.nodesource.com/node_6.x xenial main" | tee /etc/apt/sources.list.d/nodesource.list
 
-GOLANG_VERSION=1.9.4
+curl -fsSL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+
+add-apt-repository ppa:ubuntu-toolchain-r/test
+
+# cockroachdb requires gcc 6.0+
+apt-get -y update
+apt-get install -y gcc-6 g++-6 cpp-6 \
+    nodejs yarn
+
+update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-6 40
+update-alternatives --install /usr/bin/cpp cpp /usr/bin/cpp-6 40
+update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-6 40
+
+GOLANG_VERSION=1.11.5
 GOLANG_DOWNLOAD_URL=https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
-GOLANG_DOWNLOAD_SHA256=15b0937615809f87321a457bb1265f946f9f6e736c563d6c5e0bd2c22e44f779
+GOLANG_DOWNLOAD_SHA256=ff54aafedff961eb94792487e827515da683d61a5f9482f668008832631e5d25
 
 rm -rf /usr/local/go
 curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz
 echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c -
 tar -C /usr/local -xzf golang.tar.gz
 rm golang.tar.gz
 
-mkdir -p /gopath/src/github.com/cockroachdb
-mv /pkg/src/cockroach /gopath/src/github.com/cockroachdb/
+cd /pkg/src/cockroach
 
-cd /gopath/src/github.com/cockroachdb/cockroach
+# go-libedit requries `ncurses` library. We need to replace `cgo` flags so that
+# go compiler passes path to DCOS included ncurses library.
+sed -i 's/LDFLAGS: -lncurses/LDFLAGS: -L\/opt\/mesosphere\/active\/ncurses\/lib -ltinfo -lncurses/g' src/github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix/editline_unix.go
```

Comment:
You can avoid the backslashes by using a different delimiter (e.g. `s#...#...#g`)